action/protocol/staking/protocol.go

```diff
@@ -76,12 +76,18 @@ func (p *Protocol) Start(ctx context.Context) error {
 	}
 
 	// decode the candidate and put into candidate center
+	thres := big.NewInt(p.config.Register.MinSelfStake)
 	for i := 0; i < iter.Size(); i++ {
 		c := &Candidate{}
 		if err := iter.Next(c); err != nil {
 			return errors.Wrapf(err, "failed to deserialize candidate")
 		}
-		p.inMemCandidates.Put(c)
+
+		if c.SelfStake.Cmp(thres) >= 0 {
+			if err := p.inMemCandidates.Upsert(c); err != nil {
+				return err
+			}
+		}
```

Comment:
I thought no matter what, we will keep the candidate in memory to maintain unique name/operator address and etc.

Reply:
oh yeah, my bad
voter can still stake to inactive candidate (like being slashed)